accessor/cuda_helper.hpp

```diff
@@ -17,7 +17,15 @@
 #include "utils.hpp"
 
 
+struct __half;
```

Comment:
I don't think there's a way around the issue, but we should be aware that `__half` is a reserved identifier according to the C++ standard, so we should technically not be defining anything with that name. Also there might be a tiny potential of name mangling issues if this is ever handled as a `class` instead of a `struct` with MSVC.